src/hooks/Filters/filter-complete.jsx

```diff
@@ -0,0 +1,40 @@
+function clearFilterHandler(dispatch) {
+dispatch({ type: "clear" })
+}
+
+function discountHandler(e,dispatch) {
+dispatch({ type: "discount", payload: e.target.value })
+}
+
+function rating4Handler(dispatch) {
+dispatch({ type: "rating4", payload: "rating4" })
+}
+    
+function rating3Handler(dispatch) {
+dispatch({ type: "rating3", payload: "rating3" })
+}
+    
+function rating2Handler(dispatch) {
+dispatch({ type: "rating2", payload: "rating2" })
+}
+    
+function rating1Handler(dispatch) {
+dispatch({ type: "rating1", payload: "rating1" })
+}
+
```

Comment:
```
function rating4Handler(dispatch) {
dispatch({ type: "rating4", payload: "rating4" })
}

function rating3Handler(dispatch) {
dispatch({ type: "rating3", payload: "rating3" })
}

function rating2Handler(dispatch) {
dispatch({ type: "rating2", payload: "rating2" })
}

function rating1Handler(dispatch) {
dispatch({ type: "rating1", payload: "rating1" })
}
```

Suggestion: This is also fine, but these functions could be written into single function called ratinghandler and in that function you can pass the "type" and "payload" as parameter as use it